File: HackerHank/DiagonalDifference/DiagonalDifferenceSolution.py
def diagonalDifference(arr):
    sumDiag1 = 0
    sumDiag2 = 0
    for i in range(0, len(arr)):
        sumDiag1 += arr[i][i]
        sumDiag2 += arr[i][len(arr[i])-i-1]
    difference = abs(sumDiag1-sumDiag2)
    return difference

File: HackerHank/DiagonalDifference/DiagonalDifferenceSolution.cpp
#include <vector>
using namespace std;

int diagonalDifference(vector<vector<int>> arr)
{
    int sumDiagonal1 = 0;
    int sumDiagonal2 = 0;
    for (int i = 0; i < arr.size(); i++)
    {
        sumDiagonal1 += arr[i][i];
        sumDiagonal2 += arr[i][arr.size() - i - 1];
    }
    int difference = sumDiagonal1 - sumDiagonal2;
    if (difference < 0)
        difference = -difference;
    return difference;
}